src/services/leaderboard/leaderboard-utils.ts

```diff
@@ -205,6 +207,23 @@ export async function promoteUsersToNextTier(
         throw error;
     }
 
+    // Enroll every userId into a Firebase topic if they got promoted today
+    for (const userId of userIds) {
+        const { data: userDevice } = await SupabaseDB.NOTIFICATIONS.select(
+            "deviceId"
+        )
+            .eq("userId", userId)
+            .maybeSingle()
+            .throwOnError();
+        if (!userDevice?.deviceId) {
+            continue; // we can just be done here if they don't have a deviceId
+        }
+        const topicName = `tier-promotion-${day.toLowerCase()}`;
+        await getFirebaseAdmin()
+            .messaging()
+            .subscribeToTopic(userDevice.deviceId, topicName);
+    }
```

Comment:
Never for loop db calls - make this a single db call please - same with firebase admin subscribing